src/border_agent/border_agent.cpp

```diff
@@ -158,6 +158,9 @@ BorderAgent::BorderAgent(otbr::Ncp::RcpHost &aHost, Mdns::Publisher &aPublisher)
     , mBaseServiceInstanceName(OTBR_MESHCOP_SERVICE_INSTANCE_NAME)
 {
     mHost.AddThreadStateChangedCallback([this](otChangedFlags aFlags) { HandleThreadStateChanged(aFlags); });
+    mIsEphemeralKeyEnabled = (otThreadGetVersion() >= 5);
```

Comment:
As discussed offline, could you use a macro to configure whether it's enabled by default?

Reply:
Current modification will set init to false; later to add "OT_THREAD_VERSION_1_4" in check, as it's in the core now.